### PR TITLE
v5.0.0 post release chores

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

I added also backport tasks for v2.5.x, v3.4.x, v4.2.x and 5.1.x and removed the backport taks for discontinued releases.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
🙏

File: RELEASES.md [hunk 1]
```diff
@@ -72,12 +72,9 @@ Only the following major release series have a stable release status:
 |`v2.2.x`|February 01, 2023|
 |`v2.3.x`|February 01, 2023|
 |`v2.4.x`|February 01, 2023|
-|`v3.0.x`|March 15, 2023|
-|`v3.1.x`|March 15, 2023|
```
Comment (colin-axner) [inline RELEASES.md:76]: I wonder if we should note that these have been discontinued? Maybe a blanket statement like all missing minor release versions have been discontinued

**Review by colin-axner — APPROVED** [review]
Should we make adding v5 to compatibility tests a post release chore?

**Comment by crodriguezvega** [comment]
> Should we make adding v5 to compatibility tests a post release chore?

Yeah, good idea! @chatton, can you please check if the changes look ok to you?

**Review by chatton — CHANGES_REQUESTED** [review]
Thanks for doing these followups @crodriguezvega !

We need to make some changes though for the test matricies, we want to still be able to run them from the release-v5 branch. So rather than updating the existing set, we can create a new set for v6 (make sure the name of the files align as this is important for reading the correct file)

The new v6 file can run tests against v5.0.0 as well as the previous versions.

All the other changes look good!

**Comment by chatton** [comment]
@crodriguezvega it might be worth triggering the v6 test suite before you merge to main. You can do this in the UI via

`Actions` -> `Compatibility E2E` -> `Run Workflow` (choose your branch, not main) -> `select release-v6.0.x`

**Review by chatton — APPROVED** [review]
LGTM! Thanks @crodriguezvega

**Pull request merged**